src/Microsoft.Tye.Core/ProjectReader.cs

```diff
@@ -30,94 +39,136 @@ public static async Task ReadProjectDetailsAsync(OutputContext output, FileInfo
                 throw new ArgumentNullException(nameof(project));
             }
 
+            EnsureMSBuildRegistered(projectFile);
+
             using (var step = output.BeginStep("Reading Project Details..."))
             {
-                await EvaluateMSBuildAsync(output, projectFile, project);
+                EvaluateProject(output, projectFile, project);
 
                 if (!SemVersion.TryParse(project.Version, out var version))
                 {
-                    output.WriteInfoLine($"No version or invalid version 'application.Version' found, using default.");
+                    output.WriteInfoLine($"No version or invalid version '{project.Version}' found, using default.");
                     version = new SemVersion(0, 1, 0);
                     project.Version = version.ToString();
                 }
 
                 step.MarkComplete();
             }
+
+            return Task.CompletedTask;
+        }
+
+        private static void EnsureMSBuildRegistered(FileInfo projectFile)
+        {
+            if (!registered)
+            {
+                // It says VisualStudio - but we'll just use .NET SDK
+                var instances = MSBuildLocator.QueryVisualStudioInstances(new VisualStudioInstanceQueryOptions()
+                {
+                    DiscoveryTypes = DiscoveryType.DotNetSdk,
+
+                    // Using the project as the working directory. We're making the assumption that
+                    // all of the projects want to use the same SDK version. This library is going
+                    // load a single version of the SDK's assemblies into our process, so we can't
+                    // use supprt SDKs at once without getting really tricky.
```

Comment:
Comment confuses me at the end. Is the point that we can't use multiple sdks in the same process without getting really tricky?

Reply:
Correct.